File: euler/problem003/java/Main.java
/*
 *
 * The prime factors of 13195 are 5, 7, 13 and 29.
 *
 * What is the largest prime factor of the number 600851475143 ?
 *
 *
*/

import java.io.*;
import java.math.*;

public class Main {

  public static void main(String args[]){

    long largest = 0L;
    long number = 600851475143L;

    for (long i = 4L; i < number/2; i++){
      if (number%i == 0 && isPrime(i)) {
        //largest = i;
        largest = i;
        System.out.println(largest);
      }
    }
    System.out.println(largest);
  }

  public static boolean isPrime(long n){
    for (long i = 3L; i < n; i += 2){
      if (n%i == 0)
        return false;
    }
    return true;
  }
}

File: euler/problem003/cpp/main.cpp
/*
 * 
 * The prime factors of 13195 are 5, 7, 13 and 29.
 *
 * What is the largest prime factor of the number 600851475143 ?
 * 
*/
#include<iostream>
#include<string>
#include<cmath>
#include<stdlib.h>
using namespace std;


bool isPrime(long int n);

int main(){

  long double largest = 0;
  for (long int i = 1; i < 600851475143; i++){
    if (600851475143%i == 0 && isPrime(i))
      largest = i;
  }
  cout << largest << endl;
	return 0;
}

bool isPrime(long int n){
  for (int i = 2; i < n; i++){
    if (n%i == 0)
      return false;
  }
  return true;
}
